### PR TITLE
docs: deprecations usage improvements and one tech change

### PR DESCRIPTION
**Pull request by jenniferfell**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

* API subsections are H4
* Pipes section talks about removal candidate "in the next major version"
* No quick way to see what might be removed in the next major version

Issue Number: N/A

## What is the new behavior?
* API subsections are H3
* Pipes section talks about removal candidate "in version 9"
* Added index to this doc, by removal candidate version. If something was a candidate for removal in v7 or v8, rounded up to say v9 because that's the next opportunity to remove them

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by jenniferfell** [comment]
In nav, see Release Information > Deprecations

**Comment by jenniferfell** [comment]
@IgorMinar : Fixed heading levels and "in the next major version"(from your previous comments) 
@gkalpak @StephenFluin @brandonroberts  : Added index by the release in which a feature can be removed (per a couple votes from y'all on focusing on future-proofing apps as soon as practical, instead of focusing on the past or anything that encourages foot-dragging on migration)

Note: Many of the items listed as candidates for removal in v9 were actually candidates for removal in v7 and v8, but v9 is the next opportunity for removal. See comments in markup for details.

File: aio/content/guide/deprecations.md [hunk 1]
```diff
@@ -18,6 +18,50 @@ For step-by-step instructions on how to update to the latest Angular release, us
...
+| `@angular/core` | [`TestBedStatic.deprecatedOverrideProvider()`](#core-testing) | <!--v8--> v9 | 
+| `@angular/core` | [`ViewEncapsulation.Native`](#core) | v9 | 
+| `@angular/forms` | [`ngForm` element selector](#ngform) | v9 |
+| `@angular/forms` | [`NgFormSelectorWarning`](#forms) | v9 | 
+| `@angular/forms` | [`ngModel` with reactive forms](#ngmodel-reactive) | v9 | 
+| `@angular/router` | [`preserveQueryParams`](#router) | <!--v7--> v9 | 
+| `@angular/upgrade` | [`@angular/upgrade`](#upgrade) | <!--v8--> v9 | 
+| `@angular/upgrade` | [`getAngularLib`](#upgrade-static) | <!--v8--> v9 | 
+| `@angular/upgrade` | [`setAngularLib`](#upgrade-static) | <!--v8--> v9 | 
+| template syntax | [`/deep/`, `>>>`, and `::ng-deep`](#component-styles) | <!--v7--> v9 |
```
Comment (trotyl) [inline aio/content/guide/deprecations.md]: `::ng-deep` was only deprecated but never planed to be removed in future.

File: aio/content/guide/deprecations.md [hunk 2]
```diff
@@ -18,6 +18,50 @@ For step-by-step instructions on how to update to the latest Angular release, us
...
+| `@angular/router` | [`preserveQueryParams`](#router) | <!--v7--> v9 | 
+| `@angular/upgrade` | [`@angular/upgrade`](#upgrade) | <!--v8--> v9 | 
+| `@angular/upgrade` | [`getAngularLib`](#upgrade-static) | <!--v8--> v9 | 
+| `@angular/upgrade` | [`setAngularLib`](#upgrade-static) | <!--v8--> v9 | 
+| template syntax | [`/deep/`, `>>>`, and `::ng-deep`](#component-styles) | <!--v7--> v9 | 
+| template syntax | [`<template`>](#template-tag) | <!--v7--> v9 |  
+| service worker | [`versionedFiles` setting](#sw-versionedfiles)| v9 | 
+| `@angular/core` | [`TestBed.deprecatedOverrideProvider()`](#core-testing) | v10 | 
+| `@angular/core` | [`defineInjectable`](#core) | v11 | 
+| `@angular/core` | [`inject`](#core) | v11 |
```
Comment (trotyl) [inline aio/content/guide/deprecations.md]: Deprecation of `inject` was a mistake and has already been reverted: https://github.com/angular/angular/pull/30132

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (jenniferfell) [reply aio/content/guide/deprecations.md]: @IgorMinar : Need engineering confirmation. It's not clear from what is said in the docs: https://angular.io/guide/component-styles#deprecated-deep--and-ng-deep

File: aio/content/guide/deprecations.md (hunk 2 above)
Comment (jenniferfell) [reply aio/content/guide/deprecations.md]: Thanks. Removed.

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (trotyl) [reply aio/content/guide/deprecations.md]: Ref https://github.com/angular/angular/issues/17867#issuecomment-312322174

Comment (jenniferfell) [reply aio/content/guide/deprecations.md]: Related https://github.com/angular/angular/issues/25160

**Review by Splaktar — APPROVED** [review]
LGTM in general. I think that the `h4` to `h3` change helped, but I still feel like the general structure of the page needs some improvements. Perhaps just some better white space would help, but it also feels a little odd having a huge list of deprecated APIs come first and then the latest removed APIs (which are likely the most frequently updated and interesting) left to the end.

**Comment by jenniferfell** [comment]
@Splaktar Thanks for looking at this, and so quickly! I'll try to look at the white space tonight. Re the overall structure, we also have release notes for each release, which are intended to highlight exactly what is announced as deprecated or being removed in that particular release happening right now. So, I went with an org that focuses on the idea that "you need to migrate off all of these things in the next 18 months." As deprecated features are removed, they fall to the bottom of the page...or perhaps even off the page. We're not sure how far back we want this list of removals to go, but we had to have a place in AIO (not in the release note on the blog) that helped with redirects for @angular/http api pages. 

Pondering your comments more. Maybe we can slack when I'm online next week. Thanks again!

**Comment by trotyl** [comment]
> So, I went with an org that focuses on the idea that "you need to migrate off all of these things in the next 18 months." 

@jenniferfell That's why **deprecated feature cannot be removed until migration become possible**.

Component style mixin is a feature definitely needed, but the current condition is, `/deep/` is deprecated because of dropped by old web standard, the new web standard is [`::part & ::theme`](https://github.com/angular/angular/issues/22515). However, with Angular team focusing on other more important work, there's no non-deprecated way being made yet, so scheduling a removal makes no sense at all.

It's not possible for `::ng-deep` to be removed without achieving this feature is other ways, otherwise people will for sure drop Angular.

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (Splaktar) [reply aio/content/guide/deprecations.md]: Yeah, I don't think that we should mark this as scheduled for removal in Angular version 9. We need to get the [alternative feature](https://github.com/angular/angular/issues/22515) shipped first and work has not started on it.

Comment (Splaktar) [reply aio/content/guide/deprecations.md]: @StephenFluin thoughts?

**Comment by jenniferfell** [comment]
> > So, I went with an org that focuses on the idea that "you need to migrate off all of these things in the next 18 months."
> 
> @jenniferfell That's why **deprecated feature cannot be removed until migration become possible**.
> 
> Component style mixin is a feature definitely needed, but the current condition is, `/deep/` is deprecated because of dropped by old web standard, the new web standard is [`::part & ::theme`](https://github.com/angular/angular/issues/22515). However, with Angular team focusing on other more important work, there's no non-deprecated way being made yet, so scheduling a removal makes no sense at all.
> 
> It's not possible for `::ng-deep` to be removed without achieving this feature is other ways, otherwise people will for sure drop Angular.

"Candidate for removal at version x" does not mean that it will be removed, for the reason you mention. As you can see, there are features that were announced as deprecated in versions 4 and 5 but are not yet removed.

**Comment by trotyl** [comment]
> "Candidate for removal at version x" does not mean that it will be removed

Then any reason for calling it "candidate for removal at version X" if it's never designed to be removed at version X? 

> there are features that were announced as deprecated in versions 4 and 5 but are not yet removed.

"Removal delayed" and "removal not planed" are different scenarios, the former is ready to be removed and the latter is not.

**Review by IgorMinar — APPROVED** [review]
I pushed a few more commits with changes I made. quick summary:

- I renamed the heading and anchor for deep selectors
- I removed trailing whitespace
- I remove deprecatedOverrideProvider from the list to avoid a conflict with https://github.com/angular/angular/pull/30576
- I added reflect-metadata polyfill to the list of deprecated apis to match the changes being made in https://github.com/angular/angular-cli/pull/14473

**Comment by jenniferfell** [comment]
* Rebased 
* Removed other mention of deprecatedOverrideProvider from the doc. 
* Squashed commits

**Comment by IgorMinar** [comment]
this now conflicts. I'm going to rebase it.

**Comment by IgorMinar** [comment]
rebased + forcepushed

**Pull request closed without merging**